auth/resource.cc

```diff
@@ -106,6 +121,39 @@ resource::resource(role_resource_t, std::string_view role) : resource(resource_k
 resource::resource(service_level_resource_t): resource(resource_kind::service_level) {
 }
 
+resource::resource(functions_resource_t) : resource(resource_kind::functions) {
+}
+
+resource::resource(functions_resource_t, std::string_view keyspace) : resource(resource_kind::functions) {
+    _parts.emplace_back(keyspace);
+}
+
+resource::resource(functions_resource_t, std::string_view keyspace, std::string_view function_name) : resource(resource_kind::functions) {
+    _parts.emplace_back(keyspace);
+    _parts.emplace_back(function_name);
+}
+
+resource::resource(functions_resource_t, std::string_view keyspace, std::string_view function_name, std::vector<sstring> function_signature) : resource(resource_kind::functions) {
+    _parts.emplace_back(keyspace);
+    sstring encoded_signature = format("{}[{}]",
+            function_name,
```

Comment:
The workflow looks like this:

1. First we get the function name and signature string from the cql parser - if some strings were quoted in the input, they arrive unquoted from the parser
2. We encode them to the "{}[{}]" format, where the name is written unchanged before the brackets, and the types `prepare`d and then written in the brackets. Even if the arguments contained some UDTs with weird names, this is no longer a problem, because after we `prepare` them, the names are stored as bytes.
3. When we try to decode them, we look for the last `[`, because even if the function name contained `[`, no argument type has it. For the same reason we can split the arguments on '^'

So there should be no problems (as the "weird" test confirms)

Reply:
I'm not seeing it documented anywhere, but after checking manually, the UDTs names are basically stored as number strings after we `prepare` them. I should have just added a test for it now that UDTs work as well. Will send a follow-up